elasticapm/instrumentation/packages/elasticsearch.py

```diff
@@ -0,0 +1,197 @@
+            'info': ['params'],
+            'mget': ['body', 'index', 'doc_type', 'params'],
+            'msearch': ['body', 'index', 'doc_type', 'params'],
+            'msearch_template': ['body', 'index', 'doc_type', 'params'],
+            'mtermvectors': ['index', 'doc_type', 'body', 'params'],
+            'ping': ['params'],
+            'put_template': ['id', 'body', 'params'],
+            'reindex': ['body', 'params'],
+            'reindex_rethrottle': ['task_id', 'params'],
+            'render_search_template': ['id', 'body', 'params'],
+            'scroll': ['scroll_id', 'body', 'params'],
+            'search': ['index', 'doc_type', 'body', 'params'],
+            'search_shards': ['index', 'doc_type', 'params'],
+            'search_template': ['index', 'doc_type', 'body', 'params'],
+            'termvectors': ['index', 'doc_type', 'id', 'body', 'params'],
+            'update': ['index', 'doc_type', 'id', 'body', 'params'],
+            'update_by_query': ['index', 'doc_type', 'body', 'params'],
+        }
+    }
+
+    query_methods = ('search', 'count', 'delete_by_query')
+
+    instrument_list = [
+        ("elasticsearch.client", "Elasticsearch.ping"),
+        ("elasticsearch.client", "Elasticsearch.info"),
+        ("elasticsearch.client", "Elasticsearch.create"),
+        ("elasticsearch.client", "Elasticsearch.index"),
+        ("elasticsearch.client", "Elasticsearch.count"),
+        ("elasticsearch.client", "Elasticsearch.delete"),
+        ("elasticsearch.client", "Elasticsearch.delete_by_query"),
+        ("elasticsearch.client", "Elasticsearch.exists"),
+        ("elasticsearch.client", "Elasticsearch.exists_source"),
+        ("elasticsearch.client", "Elasticsearch.get"),
+        ("elasticsearch.client", "Elasticsearch.get_source"),
+        ("elasticsearch.client", "Elasticsearch.search"),
+        ("elasticsearch.client", "Elasticsearch.update"),
+
+        # TODO:
+        # ("elasticsearch.client", "Elasticsearch.update_by_query"),
+        # ("elasticsearch.client", "Elasticsearch.search_shards"),
+        # ("elasticsearch.client", "Elasticsearch.put_script"),
+        # ("elasticsearch.client", "Elasticsearch.get_script"),
+        # ("elasticsearch.client", "Elasticsearch.delete_script"),
+        # ("elasticsearch.client", "Elasticsearch.put_template"),
+        # ("elasticsearch.client", "Elasticsearch.get_template"),
+        # ("elasticsearch.client", "Elasticsearch.explain"),
+        # ("elasticsearch.client", "Elasticsearch.termvectors"),
+        # ("elasticsearch.client", "Elasticsearch.mtermvectors"),
+    ]
+
+    def __init__(self):
+        super(ElasticsearchInstrumentation, self).__init__()
+        try:
+            from elasticsearch import VERSION
+            self.version = VERSION[0]
+        except ImportError:
+            self.version = None
+
+    def instrument(self):
+        if self.version and not 2 <= self.version < 7:
```

Comment:
should instrumentation also be skipped if version is none, but with a different message?

Reply:
`version` is set to `None` only in the case of an `ImportError`. We handle that case in the super method with an appropriate log message, so I think it's ok to fall through.

https://github.com/elastic/apm-agent-python/blob/e74bd1a55b275e4d147699308c22785595c4636e/elasticapm/instrumentation/packages/base.py#L73-L77